### PR TITLE
Improve infrastructure for better integration tests

### PR DESCRIPTION
**Pull request by jsotuyod**
- Use find-sec-bugs' Hamcrest matchers for assertions
 - Deprecate the old detector based tests / annotations
 - Rewriting those as proper tests will be hard... over 1000 files,
    but only half actually use expectation annotations
 - Provides a better testing framework for #10 

I'm getting really mad at the current ant chaos... I wanted to create a separate source folder for integration tests and another for bug samples, but it was so chaotc I just gave up. Tidying it up / getting rid of it as per #4 would make my day so much happier...

### CODE REVIEW

File: findbugs/src/junit/com/h3xstream/findbugs/test/jsp/SMAPSourceDebugExtension.java [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+public class SMAPSourceDebugExtension {
+
+
+    private final Map<Integer, FileInfo> fileinfo = new HashMap<Integer, FileInfo>();
+    private final Map<Integer, List<Integer>> reverseLineMapping = new HashMap<Integer, List<Integer>>();
+
+    private static final Pattern LINE_INFO_PATTERN = Pattern.compile("([0-9]+)(?:#([0-9]+))?(?:,([0-9]+))?:([0-9]+)(?:,([0-9]+))?");
+
+    public SMAPSourceDebugExtension(String value) {
+        String[] lines = value.split("\n");
```
Comment (mebigfatguy) [inline findbugs/src/junit/com/h3xstream/findbugs/test/jsp/SMAPSourceDebugExtension.java]: maybe you want ("\\s*\n\\s*)

Comment (jsotuyod) [inline findbugs/src/junit/com/h3xstream/findbugs/test/jsp/SMAPSourceDebugExtension.java]: Not sure, copied it as is from find-sec-bugs. @h3xstream?

Comment (h3xstream) [inline findbugs/src/junit/com/h3xstream/findbugs/test/jsp/SMAPSourceDebugExtension.java]: The JSP support in the test utility need to be update to improved source line mapping. I just forgot about it.

On the other hand, I would recommend to use https://github.com/SonarQubeCommunity/sonar-findbugs/blob/master/src/main/java/org/sonar/plugins/findbugs/resource/SmapParser.java. This implementation was tested with various JSP compiler.  `BufferedReader.readLine()` takes into account the different new line patterns.

Comment (jsotuyod) [inline findbugs/src/junit/com/h3xstream/findbugs/test/jsp/SMAPSourceDebugExtension.java]: Interesting, will look into it and try to integrate it.

**Comment by jsotuyod** [comment]
There you go guys. Hopefully I understood SMAP correctly (had to check the JSR).

**Comment by jsotuyod** [comment]
@h3xstream had the chance to check this?

File: findbugs/src/junit/com/h3xstream/findbugs/test/service/ClassFileLocator.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library.
+ */
+package com.h3xstream.findbugs.test.service;
```
Comment (jsotuyod) [inline findbugs/src/junit/com/h3xstream/findbugs/test/service/ClassFileLocator.java]: Ok, didn't want to move it without your consent. Bare in mind this is in src/junit and not being published for the time being

**Comment by jsotuyod** [comment]
Thanks for the feedback, I will try to fix this later today or tomorrow.

**Comment by jsotuyod** [comment]
@h3xstream done! Let me know if you see anything else!

**Comment by h3xstream** [comment]
@jsotuyod Looks good.

**Pull request merged**